### PR TITLE
Aden 13619 2

### PR DESCRIPTION
**Pull request by adi-wesolowski-fandom**

### CODE REVIEW

File: src/ad-bidders/prebid/liveramp-ats/index.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import { UserIdConfig } from '../index';
+
+const logGroup = 'LiveRamp';
+
+export abstract class LiveRampAtsTypes {
+	static PLACEMENT_ID = '2161';
+	static ENVELOPE_STORAGE_NAME = 'idl_env';
+}
+
+class LiveRampAts {
```
Comment (mfarul) [inline src/ad-bidders/prebid/liveramp-ats/index.ts:11]: It is rather RampID setup for prebid. ATS is build on it.

File: src/ad-services/ats/index.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+			utils.logger(logGroup, 'no hashes');
+			return;
+		}
+
+		const [md5, sha1, sha256] = userEmailHashes;
+		const launchpadScript = utils.scriptLoader.loadScript(this.launchpadScriptUrl);
+		const launchpadBundleScript = utils.scriptLoader.loadScript(this.launchpadBundleScriptUrl);
+
+		return Promise.all([launchpadScript, launchpadBundleScript])
+			.then(() => this.waitForAts())
```
Comment (mfarul) [inline src/ad-services/ats/index.ts:37]: is this needed? then should be fired after the script is loaded & executed, so the variable should be there...

File: src/ad-bidders/prebid/index.ts [hunk 3]
```diff
@@ -284,6 +295,23 @@ export class PrebidProvider extends BidderProvider {
 		}
 	}
 
+	private enableATSAnalytics(): void {
+		if (context.get('bidders.liveRampATSAnalytics.enabled')) {
```
Comment (mfarul) [inline src/ad-bidders/prebid/index.ts:299]: we should also check if liveRampId is enabled, as analytics probably will not work without prebid module

**Review by mfarul — APPROVED** [review]

File: src/ad-bidders/prebid/liveramp-ats/index.ts (hunk 1 above)
Comment (preludek) [reply src/ad-bidders/prebid/liveramp-ats/index.ts]: renamed

File: src/ad-bidders/prebid/index.ts (hunk 3 above)
Comment (preludek) [reply src/ad-bidders/prebid/index.ts]: done

File: src/ad-services/ats/index.ts (hunk 2 above)
Comment (preludek) [reply src/ad-services/ats/index.ts]: I removed it and scripts are being loaded so I guess it's redundant. Removing it

**Review by preludek — APPROVED** [review]

**Pull request merged**